### PR TITLE
list filter improvements

### PR DESCRIPTION
**Pull request by tolry**
- [x] filter state should be reflected in URL
- additional filters
  - [x] title search
  - [x] at least on speaker available
  - [ ] at least x votes cast?
  - [x] filter by speaker, author, ....?
- [x] make sort order configurable
- [x] two column layout: filter left, list on the right
- [x] fully faceted search
- [x] filter presets? e.g. "most popular", "upcoming talks", "feedback needed" => could replace archive?

### CODE REVIEW

File: src/js/components/Topic/Index.jsx [hunk 1]
```diff
@@ -33,19 +35,32 @@ export default class Index extends React.Component {
         clearTimeout(this.timer);
 
         this.timer = setTimeout(() => {
-            this.setState({
-                search: value
-            });
+            this.props.location.query.search = value;
```
Comment (DavidBadura) [inline src/js/components/Topic/Index.jsx]: i think, you shouldn't modify the props or location. you should clone it before.

File: src/js/components/Topic/Index.jsx [hunk 2]
```diff
@@ -28,25 +29,48 @@ export default class Index extends React.Component {
...
         }
 
-        let data = this.state.data.filter((topic) => {
-            return topic.title.toLowerCase().includes(this.state.search.toLowerCase());
-        });
+        let filterCriteria = {
+            search: this.props.location.query.search,
+            order: this.props.location.query.order
+                ? this.props.location.query.order
+                : 'newest'
```
Comment (DavidBadura) [inline src/js/components/Topic/Index.jsx]: short version: 

```js
{
  order: this.props.location.query.order || 'newest'
}
```

File: src/js/components/Topic/Index.jsx [hunk 3]
```diff
@@ -58,21 +82,95 @@ export default class Index extends React.Component {
...
+                    : -1;
+            }
+
+            if (valueA < valueB) {
+                return direction == 'asc'
+                    ? -1
+                    : 1;
+            }
+
+            return 0;
```
Comment (DavidBadura) [inline src/js/components/Topic/Index.jsx]: i think this is equivalent 

```js
if (direction == 'asc') {
  return valueA > valueB;
}

return valueA < valueB;
```

Comment (tolry) [inline src/js/components/Topic/Index.jsx]: Wouldn't `return valueA > valueB``return boolean true|false whereas sort functions need 0|1|-1? Not sure how JS handles this exactly ...

**Comment by tolry** [comment]
@DavidBadura I am considering merging the current state, leaving cleanup and additional filters, templates  as well as design optimizations for later.

Any thoughts?

**Pull request merged**